cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/runtime/schedule/trigger/AndTrigger.java

```diff
@@ -46,4 +49,33 @@ public boolean isSatisfied(ProgramSchedule schedule, List<Notification> notifica
   public List<TriggerInfo> getTriggerInfos(TriggerInfoContext context) {
     return getUnitTriggerInfosAddRuntimeArgs(context);
   }
+
+  @Nullable
+  @Override
+  public Trigger updateTriggerWithDeletedProgram(ProgramId programId) {
```

Comment:
This is not updating the current trigger right? So don't call it `update`. Also, I think it's better to move this method to a util class (`Triggers` would be a good name).

Reply:
Hum, you actually have different implementation for AND and OR, never mind the util class part.